plugin/environment.go

```diff
@@ -488,6 +498,50 @@ func (env *Environment) UnpackWebappBundle(id string) (*model.Manifest, error) {
 	return manifest, nil
 }
 
+func patchReactDOM(b []byte) ([]byte, bool) {
+	s := string(b)
+	nameIndex := strings.Index(s, "react-dom")
```

Comment:
Changing this to `react-dom.production.min.js` is a better search, as projects that are updated to actually use React 17 now have other instances of the string `react-dom` showing up in the compiled bundle. I've confirmed that all marketplace plugins that "accidentally" bundle ReactDOM have the string `react-dom.production.min.js` present inside the function that is defining the ReactDOM package.

```suggestion
	nameIndex := strings.Index(s, "react-dom.production.min.js")
```